### DISCUSSION TITLE
bat syntax highlighting for Ghostty's config file similar to the vim plugin

### DISCUSSION THREAD

**Discussion by phillip-hirsch** [ghostty-org/ghostty#2610] [Feature Requests, Ideas] [answered]
Currently, Ghostty exports a [vim plugin](https://github.com/ghostty-org/ghostty/blob/main/src/config/vim.zig) for syntax highlighting the configuration file. However, when viewing the configuration file with [bat](https://github.com/sharkdp/bat), we lose the syntax highlighting that we have with the vim plugin. Since the vim plugin is created at compile time, we should consider doing the same for the [sublime-syntax file](https://github.com/sharkdp/bat?tab=readme-ov-file#adding-new-syntaxes—language-definitions). I have created a local branch containing changes to introduce this functionality and have attached a screenshot of the output below:

<img width="1824" alt="Screenshot 2024-11-06 at 3 37 55 PM" src="https://github.com/user-attachments/assets/ad4b5390-cf24-45c7-8dcf-e4c563280bd1">

**Comment by mitchellh** [comment]
I'm supportive of this as long as it is all automatic and we don't need any additional maintenance burden.

**Reply by phillip-hirsch** [reply]
I completely understand! My goal with this is to make it completely automated once it's been added.

**Reply by mitchellh** [reply, marked as answer]
Sounds good. Feel free to open a direct PR and you can reference this discussion. I don't think we need to make an issue to track this separately. Thanks.

**Reply by phillip-hirsch** [reply]
Thank you!
